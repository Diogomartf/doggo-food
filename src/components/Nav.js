import React from 'react';
import styled from 'styled-components';

const Nav = () => (
  <nav>
    <NavCenter>
      <Logo>Doggo Food</Logo>
    </NavCenter>
  </nav>
);

const NavCenter = styled.div`
  height: 64px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Logo = styled.div`
  font-family: Inter;
  font-style: normal;
  font-weight: 900;
  font-size: 28px;
  text-align: center;

  color: ${({ theme }) => theme.colors.primary};
`;
Logo.displayName = 'Logo';

export default Nav;
