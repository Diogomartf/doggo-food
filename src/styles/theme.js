import colors from './colors';

export default {
  name: 'default theme',
  colors
};
