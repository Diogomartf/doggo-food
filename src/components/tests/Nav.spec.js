import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import theme from '../../styles';

import Nav from '../Nav';

configure({ adapter: new Adapter() });

describe('<Nav/>', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Nav />);

    expect(wrapper).toMatchSnapshot();
  });

  it('render with correct styles', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Nav />
      </ThemeProvider>
    );

    expect(wrapper.find('Logo')).toHaveStyleRule(
      'color',
      `${theme.colors.primary}`
    );

    expect(wrapper.prop('theme')).toEqual(theme);
  });
});
