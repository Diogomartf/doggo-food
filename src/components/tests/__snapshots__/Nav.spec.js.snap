// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> should render correctly 1`] = `
<nav>
  <styled.div>
    <Logo>
      Doggo Food
    </Logo>
  </styled.div>
</nav>
`;
